backend/core/socketio_config.py

```diff
@@ -21,10 +29,23 @@
     socketio_server=sio_server, socketio_path=SOCKETIO_PATH
 )
 
+cookie = SimpleCookie()
+db_session = Session(engine)
+
 
 @sio_server.on("connect")
 async def sio_connect(sid, environ):
-    logger.info(f"A user sid[{sid}] connected")
+    try:
+        cookie.load(environ["HTTP_COOKIE"])
+        auth_token = cookie["AUTH_TOKEN"].value
+        user_phone_number = verify_jwt_token(auth_token).get(
+            "uphone_number", None
+        )
+        async with sio_server.session(sid) as session:
+            session["user_phone_number"] = user_phone_number
```

Comment:
In this code, I try to ged the token from HTTP_COOKIE instead of send the token as a parameter, then decode the token to get the user phone number. After we get the user phone number, we can put that inside the socketio session to be use later in socketio event.